From 1e4460150b888802908084ee4a478dbaee7cfea2 Mon Sep 17 00:00:00 2001
From: Arnaud Pouliquen <arnaud.pouliquen@foss.st.com>
Date: Mon, 26 Jun 2023 16:32:21 +0200
Subject: [PATCH 0487/1141] dt-bindings: remoteproc: st,stm32-rproc: Introduce
 nvmem properties

For the stm32MP25 series, add the nvmem properties to manage the access
of the TAMP registers that can contains information about the
resource table.

Signed-off-by: Arnaud Pouliquen <arnaud.pouliquen@foss.st.com>
Change-Id: I472dbf5978db630ab031bcc85da1d049e539bbc8
Reviewed-on: https://gerrit.st.com/c/mpu/oe/st/linux-stm32/+/326924
Reviewed-by: Arnaud POULIQUEN <arnaud.pouliquen@st.com>
Tested-by: Arnaud POULIQUEN <arnaud.pouliquen@st.com>
Domain-Review: Arnaud POULIQUEN <arnaud.pouliquen@st.com>
ACI: CIBUILD <MDG-smet-aci-builds@list.st.com>
ACI: CITOOLS <MDG-smet-aci-reviews@list.st.com>
---
 .../bindings/remoteproc/st,stm32-rproc.yaml           | 11 +++++++++++
 1 file changed, 11 insertions(+)

diff --git a/Documentation/devicetree/bindings/remoteproc/st,stm32-rproc.yaml b/Documentation/devicetree/bindings/remoteproc/st,stm32-rproc.yaml
index af8944977f58..a7ea628a5047 100644
--- a/Documentation/devicetree/bindings/remoteproc/st,stm32-rproc.yaml
+++ b/Documentation/devicetree/bindings/remoteproc/st,stm32-rproc.yaml
@@ -131,6 +131,17 @@ properties:
           - description: The offset of the tamp register
           - description: The field mask of the Cortex-M4 resource table address
 
+  nvmem-cells:
+    maxItems: 2
+    description: |
+      Specify the resource table address and size for firmware started by
+      the boot loader.
+
+  nvmem-cell-names:
+    items:
+      - const: rsc-tbl-addr
+      - const: rsc-tbl-size
+
   st,auto-boot:
     $ref: /schemas/types.yaml#/definitions/flag
     description:
-- 
2.39.2

